sources/packages/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -193,33 +198,54 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
    */
   async updateProgramInfo(
     assessmentId: number,
+    auditUserId: number,
     locationId: number,
     status: ProgramInfoStatus,
     programId?: number,
     offeringId?: number,
   ): Promise<StudentAssessment> {
-    const assessment = await this.repo
-      .createQueryBuilder("assessment")
-      .select(["assessment.id", "application.id"])
-      .innerJoin("assessment.application", "application")
-      .where("assessment.id = :assessmentId", { assessmentId })
-      .getOne();
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      const assessmentRepo =
+        transactionalEntityManager.getRepository(StudentAssessment);
+      const assessment = await assessmentRepo
+        .createQueryBuilder("assessment")
+        .select(["assessment.id", "application.id", "student.id"])
+        .innerJoin("assessment.application", "application")
+        .innerJoin("application.student", "student")
+        .where("assessment.id = :assessmentId", { assessmentId })
+        .getOne();
 
-    if (!assessment) {
-      throw new CustomNamedError(
-        `Not able to find the assessment id ${assessmentId}`,
-        ASSESSMENT_NOT_FOUND,
-      );
-    }
+      if (!assessment) {
+        throw new CustomNamedError(
+          `Not able to find the assessment id ${assessmentId}`,
+          ASSESSMENT_NOT_FOUND,
+        );
+      }
 
-    assessment.application.location = { id: locationId } as InstitutionLocation;
-    assessment.application.pirProgram = { id: programId } as EducationProgram;
-    assessment.application.pirStatus = status;
-    assessment.offering = {
-      id: offeringId,
-    } as EducationProgramOffering;
+      const auditUser = { id: auditUserId } as User;
+      assessment.application.location = {
+        id: locationId,
+      } as InstitutionLocation;
+      assessment.application.pirProgram = { id: programId } as EducationProgram;
+      assessment.application.pirStatus = status;
+      assessment.application.modifier = auditUser;
+      assessment.modifier = auditUser;
+      assessment.offering = {
+        id: offeringId,
+      } as EducationProgramOffering;
 
-    return this.repo.save(assessment);
+      // If the offering will be set in the assessment check for possible SIN restrictions.
+      if (offeringId) {
+        await this.studentRestrictionService.assessSINRestrictionForOfferingId(
```

Comment:
This is the new method, the others changes are due to the addition of the transaction only.